modules/cloud-function-v2/main.tf

```diff
@@ -155,12 +138,33 @@ resource "google_cloudfunctions2_function" "function" {
 }
 
 resource "google_cloudfunctions2_function_iam_binding" "default" {
-  for_each       = local.iam
+  for_each       = var.iam
   project        = var.project_id
   location       = google_cloudfunctions2_function.function.location
   cloud_function = google_cloudfunctions2_function.function.name
   role           = each.key
-  members        = each.value
+  members = (
+    role != "roles/run.invoker" || !var.trigger_config.service_account_create
+    ? each.value
+    # if invoker role is present and we create trigger sa, add it as member
+    : concat(
+      each.value, ["serviceAccount:${google_service_account.trigger_service_account[0].email}"]
+    )
+  )
+}
+
+resource "google_cloudfunctions2_service_iam_member" "member" {
```

Comment:
I don't think this resource exists. Shouldn't it be google_cloud_run_service_iam_member?

Reply:
this is the cloud function module